include/exec/task.hpp

```diff
@@ -82,6 +103,72 @@ namespace exec {
     template <class _Ty>
     using default_task_context = __default_task_context_impl;
 
+    // This is another context that might be associated with a basic_task's promise type.
+    // It is used to implement the scheduler affinity feature.
+    class __scheduler_affine_task_context_impl {
```

Comment:
Scheduler affinity should be part of the default behavior of `task`. So integrate these changed directly into the `__default_task[...]` stuff.